lib/sql.js

```diff
@@ -1185,8 +1185,22 @@ SQLConnector.prototype.buildOrderBy = function(model, order) {
     return '';
   }
   var self = this;
-  if (typeof order === 'string') {
-    order = [order];
+  var usedConnector = this.dataSource.connector.name;
+
+  if (order === 'rand') {
+    if (usedConnector === 'mysql') {
+      return 'ORDER BY RAND()';
+    } else if (usedConnector === 'postgres') {
+      return 'ORDER BY RANDOM()';
+    } else if (usedConnector === 'oracle') {
+      return 'ORDER BY dbms_random.value';
+    } else if (usedConnector === 'mssql') {
+      return 'ORDER BY NEWID()';
+    }
+  } else {
+    if (typeof order === 'string') {
+      order = [order];
+    }
```

Comment:
I fully agree with Raymond, database-specific stuff should go to the appropriate connector; `loopback-connector` should provide only the generic functionality that can be later templated/customized by individual connectors.

Look at it from this angle: what if there is a community-maintained SQL connector, how are they going to add support for `order: 'rand'`, when they cannot control `loopback-connector`?